### PR TITLE
CIV-13608 Error messages for additional application

### PR DESCRIPTION
**Pull request by manish14836**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-13608

### Change description ###

Error messages for additional application

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/routes/features/generalApplication/applicationTypeController.ts [hunk 1]
```diff
@@ -57,4 +66,28 @@ applicationTypeController.post(APPLICATION_TYPE_URL, (async (req: AppRequest | R
   }
 }) as RequestHandler);
 
+export const additionalApplicationErrorMessages: Partial<{ [key in ApplicationTypeOption]: string; }> = {
```
Comment (laarmada) [inline src/main/routes/features/generalApplication/applicationTypeController.ts:69]: why we have in the controller file function or utils? we should move to other file or the related service one.

File: src/main/modules/i18n/locales/en.json [hunk 2]
```diff
@@ -2191,7 +2191,7 @@
         "ASK_CHANGE_CLAIM_DESCRIPTION": "This is also known as an application to amend a statement of case.",
         "ASK_CHANGE_HEARING": "Ask to change a hearing date",
         "ASK_CHANGE_HEARING_DESCRIPTION": "This is also known as an application to adjourn a hearing.",
-        "ASK_DO_SOMETHING": "Ask the court to do something that's not on this list",
+        "ASK_DO_SOMETHING": "Ask the court to do:wqa:qa something that's not on this list",
```
Comment (GorilaNaranja) [inline src/main/modules/i18n/locales/en.json:2194]: Why you need to use :wqa:qa?

Comment (GorilaNaranja) [reply src/main/modules/i18n/locales/en.json:2194]: Should we use it as well in cy.json?

File: src/main/routes/features/generalApplication/addAnotherApplicationController.ts [hunk 3]
```diff
@@ -36,7 +36,9 @@ addAnotherApplicationController.post(GA_ADD_ANOTHER_APPLICATION_URL, async (req:
     if (form.hasErrors()) {
       await renderView(req, res, form);
     } else {
-      res.redirect('test');
+      const claimId = req.params.id;
+      const redirectUrl = APPLICATION_TYPE_URL.replace(':id', claimId);
```
Comment (GorilaNaranja) [inline src/main/routes/features/generalApplication/addAnotherApplicationController.ts:40]: use constructResponseUrlWithIdParams()

Comment (manish14836) [reply src/main/routes/features/generalApplication/addAnotherApplicationController.ts]: Done

File: src/main/modules/i18n/locales/en.json (hunk 2 above)
Comment (manish14836) [reply src/main/modules/i18n/locales/en.json]: It was added by mistake. Removed

File: src/main/routes/features/generalApplication/applicationTypeController.ts (hunk 1 above)
Comment (manish14836) [reply src/main/routes/features/generalApplication/applicationTypeController.ts]: Moved to generalApplicationService

**Review by GorilaNaranja — APPROVED** [review]

File: src/main/routes/features/generalApplication/checkAnswersController.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    } else {
+      await saveStatementOfTruth(redisKey, statementOfTruth);
+      res.redirect(getRedirectUrl(claimId, claim));
+    }
+  } catch (error) {
+    next(error);
+  }
+}) as RequestHandler);
+
+function getRedirectUrl(claimId: string, claim: Claim): string {
```
Comment (laarmada) [inline src/main/routes/features/generalApplication/checkAnswersController.ts:63]: please extract this to the related service.

File: src/main/routes/features/generalApplication/checkAnswersController.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  if (claim.generalApplication?.applicationTypes?.length === 1 && claim.generalApplication.applicationTypes[0].option === ApplicationTypeOption.ADJOURN_HEARING
+    && hearingMoreThan14DaysInFuture(claim)) {
+    return constructResponseUrlWithIdParams(claimId, GENERAL_APPLICATION_CONFIRM_URL);
+  } else {
+    return 'test'; // TODO: correct URL
+  }
+}
+
+function hearingMoreThan14DaysInFuture(claim: Claim): boolean {
```
Comment (laarmada) [inline src/main/routes/features/generalApplication/checkAnswersController.ts:72]: please extract this to the related service.

**Review by laarmada — CHANGES_REQUESTED** [review]

**Review by laarmada — DISMISSED** [review]

**Review by GorilaNaranja — DISMISSED** [review]

**Pull request merged**